Source/BugsnagConfiguration.m

```diff
@@ -185,10 +185,27 @@ - (void)setUserMetadataFromUser:(BugsnagUser *)user {
     [self.metadata addAttribute:BSGKeyEmail withValue:user.emailAddress toTabWithName:BSGKeyUser];
 }
 
+// =============================================================================
+// MARK: - onSendBlock
+// =============================================================================
+
 - (void)addOnSendBlock:(BugsnagOnSendBlock)block {
     [(NSMutableArray *)self.onSendBlocks addObject:[block copy]];
 }
 
+- (void)removeOnSendBlock:(BugsnagOnSendBlock _Nonnull )block
+{
+    [(NSMutableArray *)self.onSendBlocks removeObject:block];
```

Comment:
This is more for my curiosity, but I'd be interested to know how this works, as `addOnSendBlock` makes a copy of the block whereas this is passed the original parameter. The test covering this scenario does seem to pass however.

Reply:
I’ve done a little digging; I think I understand it well enough to explain.  Failing that there’s a link at the end, and please correct me if I’m wrong.  Blocks are created initially on the stack, unlike other ObjC objects. The copy effectively _moves_ it to the heap, allowing it to be retained and used after the enclosing scope has exited.  The object pointer is updated.  So the usage in `remove()` will remove the correct - heap-stored - block from the array.  http://www.friday.com/bbum/2009/08/29/blocks-tips-tricks/